python/packages/aisearch/agent_framework_aisearch/_search_provider.py

```diff
@@ -0,0 +1,850 @@
+        endpoint = self.endpoint or self.search_endpoint
+        index_name = self.index_name or self.search_index_name
+
+        if not endpoint:
+            raise ValueError("endpoint is required (or set AZURE_SEARCH_ENDPOINT)")
+        if not index_name:
+            raise ValueError("index_name is required (or set AZURE_SEARCH_INDEX_NAME)")
+
+        # Update main fields from aliases
+        self.endpoint = endpoint
+        self.index_name = index_name
+        self.api_key = self.api_key or self.search_api_key
+
+
+class AzureAISearchContextProvider(ContextProvider):
+    """Azure AI Search Context Provider with hybrid search and semantic ranking.
+
+    This provider retrieves relevant documents from Azure AI Search to provide context
+    to the AI agent. It supports two modes:
+
+    - **semantic** (default): Fast hybrid search combining vector and keyword search
+      with semantic reranking. Suitable for most RAG use cases.
+    - **agentic**: Slower multi-hop reasoning across documents using Knowledge Bases.
+      Use only for complex queries requiring cross-document reasoning.
+
+    Examples:
+        Using Settings class with environment variables:
+
+        .. code-block:: python
+
+            from agent_framework import ChatAgent
+            from agent_framework_azure_ai import AzureAIAgentClient
+            from agent_framework.azure import AzureAISearchContextProvider, AzureAISearchSettings
+            from azure.identity.aio import DefaultAzureCredential
+
+            # Load from environment variables
+            settings = AzureAISearchSettings()
+            search_provider = AzureAISearchContextProvider(settings=settings, credential=DefaultAzureCredential())
+
+        Semantic hybrid search with API key:
+
+        .. code-block:: python
+
+            # Direct API key string (converted to AzureKeyCredential automatically)
+            search_provider = AzureAISearchContextProvider(
+                endpoint="https://mysearch.search.windows.net",
+                index_name="my-index",
+                credential="my-api-key",  # String converted to AzureKeyCredential
+                mode="semantic",
+            )
+
+        Agentic retrieval for complex queries:
+
+        .. code-block:: python
+
+            # Use agentic mode for multi-hop reasoning (slower)
+            search_provider = AzureAISearchContextProvider(
+                endpoint="https://mysearch.search.windows.net",
+                index_name="my-index",
+                credential=DefaultAzureCredential(),
+                mode="agentic",  # Multi-hop reasoning
+                azure_ai_project_endpoint="https://myproject.services.ai.azure.com",
+                model_deployment_name="gpt-4o",
+                knowledge_base_name="my-knowledge-base",  # Required for agentic mode
+            )
+    """
+
+    _DEFAULT_SEARCH_CONTEXT_PROMPT = "Use the following context to answer the question:"
+
+    def __init__(
```

Comment:
instead of all the comments below, you can also replicate the way we use settings in other clients, such as here: https://github.com/microsoft/agent-framework/blob/61dbacd6f8bdf9a484e0f1858d09bd7cfb42135a/python/packages/azure-ai/agent_framework_azure_ai/_chat_client.py#L118 especially note the variables (including env_file_path) and the docstrings